src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+              error: () => {
+                handleStateErrors(NO_RECORDINGS_MESSAGE);
+              },
+            })
+        );
+      }
+    },
+    [
+      addSubscription,
+      context.reports,
+      categorizeEvaluation,
+      queryArchivedRecordings,
+      handleArchivedRecordings,
+      handleStateErrors,
+      setUsingCachedReport,
+      setReportTime,
+    ]
+  );
+
+  const generateReport = React.useCallback(() => {
+    addSubscription(
+      context.target.target().subscribe((target) => {
+        handleLoading();
+        setTargetConnectURL(target.connectUrl);
+        dispatch(automatedAnalysisAddTargetIntent(target.connectUrl));
+        // query for designated automated-analysis profiling recording
+        addSubscription(
+          queryActiveRecordings(target.connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.targetNodes[0].recordings.active.data[0] as Recording),
+              tap((recording) => {
+                if (recording == null) {
+                  throw new Error(NO_RECORDINGS_MESSAGE);
+                }
+              })
+            )
+            .subscribe({
+              next: (recording) => {
+                context.reports
+                  .reportJson(recording, target.connectUrl)
+                  .pipe(first())
+                  .subscribe({
+                    next: (report) => {
+                      categorizeEvaluation(report);
+                    },
+                    error: (_) => {
+                      handleStateErrors(FAILED_REPORT_MESSAGE);
+                    },
+                  });
+              },
+              error: (_) => {
+                // try generating report on cached or archived recordings
+                handleEmptyRecordings(target.connectUrl);
+              },
+            })
+        );
+      })
+    );
+  }, [
+    addSubscription,
+    dispatch,
+    context.api,
+    context.target,
+    context.reports,
+    automatedAnalysisAddTargetIntent,
+    setTargetConnectURL,
+    categorizeEvaluation,
+    queryActiveRecordings,
+    handleEmptyRecordings,
+    handleLoading,
+    handleStateErrors,
+  ]);
+
+  const startProfilingRecording = React.useCallback(() => {
+    const config = context.settings.automatedAnalysisRecordingConfig();
+    const attributes = automatedAnalysisConfigToRecordingAttributes(config);
+    addSubscription(
+      context.api.createRecording(attributes).subscribe((resp) => {
+        if (resp && (resp.ok || resp.status === 400)) {
+          // in-case the recording already exists
+          generateReport();
+        } else {
+          handleStateErrors(RECORDING_FAILURE_MESSAGE);
+        }
+      })
+    );
+  }, [
+    addSubscription,
+    context.api,
+    context.settings,
+    context.settings.automatedAnalysisRecordingConfig,
+    generateReport,
+    handleStateErrors,
+  ]);
+
+  const handleErrorView = React.useCallback((): [string, undefined | (() => void)] => {
```

Comment:
Maybe rename it to `getMessageAndRetry` or something similar starting with `get*` since this is for getting errorMessage and retry callback for it.

This function can just take int the errorMessage  as the argument. This way, it does not have to change in response to the error.